src/api/callback.cc

```diff
@@ -97,18 +97,22 @@ void InternalCallbackScope::Close() {
   if (closed_) return;
   closed_ = true;
 
-  Isolate* isolate = env_->isolate();
-  auto idle = OnScopeLeave([&]() { isolate->SetIdle(true); });
+  // This function must ends up with either cleanup the
+  // async id stack or pop the topmost one from it
 
-  if (!env_->can_call_into_js()) return;
   auto perform_stopping_check = [&]() {
-    if (env_->is_stopping()) {
+    if (!env_->can_call_into_js()) {
```

Comment:
The difference here between `!env_->can_call_into_js()` and `env_->is_stopping()` is trivial. `Environment::can_call_into_js()` also checks if `Environment::is_stopping_` is set. Maybe @addaleax can chime in here?

If this line is updated, the lambda variable should also be renamed as `perform_call_js_check` or similar. Also, I'd find the checks `if (!env_->can_call_into_js()) return;` below can be merged with this lambda call too.

Reply:
Yeah, semantically, this should be checking for whether we’re currently stopping the current environment through a termination exception, so `is_stopping()` would seem to be the correct choice?